src/puppetlabs/puppetdb/anonymizer.clj

```diff
@@ -17,25 +18,15 @@
    (contains? edge "target")
    (contains? edge "relationship")))
 
-(defn resource-event?
+(pls/defn-validated resource-event? :- report/resource-event-schema
   "Returns true if it looks like a resource event"
   [event]
-  (and
-   (map? event)
-   (contains? event "status")
-   (contains? event "timestamp")
-   (contains? event "resource_title")
-   (contains? event "property")
-   (contains? event "message")
-   (contains? event "new_value")
-   (contains? event "old_value")
-   (contains? event "resource_type")))
-
-(defn report?
+  (keywordize-keys event))
```

Comment:
So is this a predicate that munges the keys of an event and returns it? Seems like the function name no longer matches it's body.